packages/nx/src/plugins/js/lock-file/pnpm-parser.ts

```diff
@@ -179,7 +197,11 @@ function getNodes(
     if (!originalPackageName) {
       continue;
     }
-    const hash = createHashFromSnapshot(snapshot);
+
+    // Compute hash once, including patch hash if available
+    const patchHash = patchHashes.get(originalPackageName);
```

Comment:
`patchedDependencies` keys can be package names with an exact version, a version range, or just the name. Using `originalPackageName` here would handle the case where the key is just the package name, but it wouldn't handle the other cases. We should try to find the potential patch hash with the package name (what you have here) and with the unparsed `key`, which contains the version, and the format will match according to the lockfile version.

Reply:
Ah of course, I should have checked that instead of assuming everyone did it like us.

I've updated the code to match the same priority order pnpm uses which is I think the most accurate for complex cases. It adds a bit to the code. As an alternative, if we wanted a less complex, but more conservative approach...we could also ignore the version stuff and just match any base package (essentially as before). It'd be less accurate, but might keep the codebase more comprehensible. Up to you! Should have two working options now :)